src/opm/parser/eclipse/EclipseState/Schedule/Well/Well.cpp

```diff
@@ -812,8 +812,10 @@ bool Well::getAllowCrossFlow() const {
 }
 
 double Well::getRefDepth() const {
-    if (!this->ref_depth.has_value())
-        throw std::logic_error(fmt::format("Well: {} - tried to access not initialized well reference depth", this->name()));
+    if (!this->ref_depth)
+        // for wells with default bhp reference depth while no connections defined,
+        // using this specific value following commercial simulation reference result
+        return -1.e+20;
```

Comment:
The callers of this function expects SI values and there are at least some callers&mdash;notably in the output/reporting code&mdash;that will convert the value to output units for display/output purposes.  A bare `-1.0e+20` will coincidentally do the right thing for METRIC and PVT-M units but will be wrong in LAB and FIELD units. You will also have to look at the callers and add special casing for negative return values as needed to ensure we always output the exact value `-1.0e+20` regardless of unit system.

Reply:
Not sure what behavoir we should expect under different unit system. Created a test case https://github.com/OPM/opm-common/issues/2442 , hopefully we can figure it out with a test case.

Reply:
> Not sure what behaviour we should expect under different unit system.

The constant `-1.0e+20` is ECLIPSE's typical sentinel value for "not defined" quantities.  In other words, if ECLIPSE outputs `-1.0e+20` metres in METRIC units, then we should output `-1.0e+20` feet in FIELD units and `-1.0e+20` centimetre in LAB units.  In other words, the constant `-1.0e+20` must **not** be subject to unit conversion, meaning we must insert special case code to intercept (large) negative values and pass those directly on to the output destination.